public/script.js

```diff
@@ -6821,9 +6833,24 @@ export async function saveSettings(type) {
     });
 }
 
-export function setGenerationParamsFromPreset(preset) {
+/**
+ * Sets the generation parameters from a preset object.
+ * @param {{ genamt?: number, max_length?: number }} preset Preset object
+ * @param {boolean?} keepContextLock If true, will not unlock context if it needs to be unlocked
+ */
+export function setGenerationParamsFromPreset(preset, keepContextLock = false) {
     const needsUnlock = (preset.max_length ?? max_context) > MAX_CONTEXT_DEFAULT || (preset.genamt ?? amount_gen) > MAX_RESPONSE_DEFAULT;
-    $('#max_context_unlocked').prop('checked', needsUnlock).trigger('change');
+    if (!keepContextLock || $('#max_context_unlocked').prop('checked')) {
+        $('#max_context_unlocked').prop('checked', needsUnlock).trigger('change');
+    } else if (needsUnlock) {
+        // cap values
```

Comment:
It may actually make more sense to simply unlock and raise the limit when the backend supports it. In fact, I don't even know why the lock is there in the first place. It also has an unnecessarily scary tooltip IMO.

Reply:
Good backends will usually trim submitted contexts that are too long, so you don't have to worry about "overflowing". 

But a UI level context size lock can have some good reasons. For example, if the UI knows the context is locked to 2k, and your character card Persona is 1k tokens long, then it would be detrimental to allow the convo to exceed 1k tokens since it will start eating into the persona.

Reply:
I was referring to the "prevent context from going > 8192" checkbox.

The behavior before I tweaked it was to automatically unlock it when the context went above 8192. I added code that prevents this, thinking there may be a reason to stay at 8192 tokens for some users, but after thinking about it, I realize this is probably not the case at all. So, if the backend supports >8k, there's really no reason for Silly to stay at 8k, so I think I should just remove the keep_lock stuff.

Does that align with what you're saying?